src/mlpack/tests/lmnn_test.cpp

```diff
@@ -165,16 +162,16 @@ BOOST_AUTO_TEST_CASE(LMNNInitialGradientTest)
   lmnnfn.Gradient(coordinates, gradient);
 
   // Result calculated by hand.
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.288, 1e-5);
-  BOOST_REQUIRE_SMALL(gradient(1, 0), 1e-5);
-  BOOST_REQUIRE_SMALL(gradient(0, 1), 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 12.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.288).epsilon(1e-7));
```

Comment:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.288).epsilon(1e-7));
```

Remove the extra space here.